modules/xmpp/JingleSessionPC.js

```diff
@@ -372,6 +453,53 @@ export default class JingleSessionPC extends JingleSession {
     }
 
     /**
+     * {@inheritDoc}
+     */
+    addIceCandidates(elem) {
+        if (this.peerconnection.signalingState === 'closed') {
+            logger.warn('Ignored add ICE candidate when in closed state');
+
+            return;
+        }
+
+        // NOTE operates on each content element, can't use () =>
+        elem.each((contentIdx, content) => {
+            $(content).find('transport>candidate')
+            .each((idx, candidate) => {
+                let line = SDPUtil.candidateFromJingle(candidate);
+
+                line = line.replace('\r\n', '').replace('a=', '');
+
+                // FIXME this code does not care to handle non-bundle transport
+                const rtcCandidate = new RTCIceCandidate({
+                    sdpMLineIndex: 0,
+
+                    // FF comes up with more complex names like audio-23423,
+                    // Given that it works on both Chrome and FF without
+                    // providing it, let's leave it like this for the time being
+                    // sdpMid: 'audio',
+                    candidate: line
+                });
+
+                // Will delay the addition until the remoteDescription is set
+                if (this.peerconnection.remoteDescription.sdp) {
+                    logger.debug(`Trying to add ICE candidate: ${line}`);
+                    this.peerconnection.addIceCandidate(
+                        rtcCandidate,
+                        () => logger.debug(`addIceCandidate ok: ${line}`),
+                        error => {
+                            logger.error(
+                                `addIceCandidate failed: ${line}`, error);
+                        });
+                } else {
+                    logger.debug(`Delaying ICE candidate: ${line}`);
+                    this.candidiates.push(rtcCandidate);
```

Comment:
It's that the invitee will send 'session-initiate' and then immediately do a burst of ice candidates. There's no guarantee on how fast and when we do the initial sRD/sLD, so they have to be delayed/stored.